gutenberg.php

```diff
@@ -161,3 +161,20 @@ function gutenberg_rest_nonce() {
 	exit( wp_create_nonce( 'wp_rest' ) );
 }
 add_action( 'wp_ajax_gutenberg_rest_nonce', 'gutenberg_rest_nonce' );
+
+
+/**
+ * Exposes the site icon url to the Gutenberg editor through the WordPress REST API.
+ */
+function register_site_icon_url_setting() {
+	register_setting(
+		'general',
+		'site_icon_url',
+		array(
+			'default'      => get_site_icon_url(),
+			'show_in_rest' => true,
+		)
+	);
+}
+
+add_action( 'rest_api_init', 'register_site_icon_url_setting' );
```

Comment:
Let's make sure we follow-up with a trac ticket here. cc @TimothyBJacobs

Reply:
Also what is the permissions situation with this setting? Since it's only for display we should ensure it's accessible to all roles.

Reply:
Thanks for calling out permissions @mtias . All roles that can access the editor but that cannot edit site configuration (Contributor, Author, and Editor) receive a 403 response when querying site information. No user uploaded site icon is displayed except for Admins and Super Admins.

Correct me if I'm wrong, but it seems like exposing site icon Url through the WordPress REST API isn't possible unless we alter the permissions of the `wp/v2/settings` endpoint. This seems like a non-starter. 

@noahtallen and I had discussed other approaches. For example, instead of using the REST API, we might inject site icon Url as a prop on plugin initialization. Let me know if you have other ideas.

Reply:
I implemented the `rest_index` filter. The site icon is now viewable by all roles (except subscriber, which can't access the editor)